src/rgw/rgw_http_client.cc

```diff
@@ -294,6 +294,51 @@ void RGWIOProvider::assign_io(RGWIOIDProvider& io_id_provider, int io_type)
   }
 }
 
+RGWHTTPClient::RGWHTTPClient(CephContext *cct,
+                             const string& _method,
+                             const string& _url)
+    : NoDoutPrefix(cct, dout_subsys),
+      has_send_len(false),
+      http_status(HTTP_STATUS_NOSTATUS),
+      req_data(nullptr),
+      verify_ssl(cct->_conf->rgw_verify_ssl),
+      cct(cct),
+      method(_method),
+      url(_url) {
+  init();
+}
+
+std::ostream& RGWHTTPClient::gen_prefix(std::ostream& out) const
+{
+  out << "http_client[" << method << "/" << url << "]";
```

Comment:
this is a neat idea! maybe `method << " " << url` so we see `GET /admin/log`?

using keystone auth as an example, we're running RGWHTTPClient in the context of a request, so we want to see the prefix from its req_state and op too

i'd suggest using `DoutPrefixPipe` instead, so you can wrap the caller's dpp and use `add_prefix()` to add this extra output

you should also update the `dout()`s to `ldpp_dout()`s in the class so we actually see this